File: smoke_tests/swift/weak_crypto.swift
// Weak Cryptography vulnerabilities in Swift
import Foundation
import CommonCrypto

class WeakCryptoVulnerabilities {
    func hashMd5(input: String) -> String {
        // VULNERABLE: MD5 is cryptographically broken
        let data = input.data(using: .utf8)!
        var digest = [UInt8](repeating: 0, count: Int(CC_MD5_DIGEST_LENGTH))
        data.withUnsafeBytes {
            _ = CC_MD5($0.baseAddress, CC_LONG(data.count), &digest)
        }
        return digest.map { String(format: "%02x", $0) }.joined()
    }

    func hashSha1(input: String) -> String {
        // VULNERABLE: SHA1 is deprecated
        let data = input.data(using: .utf8)!
        var digest = [UInt8](repeating: 0, count: Int(CC_SHA1_DIGEST_LENGTH))
        data.withUnsafeBytes {
            _ = CC_SHA1($0.baseAddress, CC_LONG(data.count), &digest)
        }
        return digest.map { String(format: "%02x", $0) }.joined()
    }

    func generateToken() -> Int {
        // VULNERABLE: Non-cryptographic random
        return Int.random(in: 0..<1000000)
    }

    func weakSessionId() -> String {
        // VULNERABLE: Predictable session ID
        return String(Date().timeIntervalSince1970)
    }

    func weakRandom() -> UInt32 {
        // VULNERABLE: arc4random without bounds
        return arc4random()
    }
}

File: smoke_tests/cpp/weak_crypto.cpp
// Weak Cryptography vulnerabilities in C++
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>
#include <openssl/md5.h>
#include <openssl/des.h>

std::string hash_md5(const std::string& input) {
    // VULNERABLE: MD5 is cryptographically broken
    unsigned char digest[MD5_DIGEST_LENGTH];
    MD5((unsigned char*)input.c_str(), input.length(), digest);
    return std::string((char*)digest, MD5_DIGEST_LENGTH);
}

void encrypt_des(const std::string& data, const std::string& key) {
    // VULNERABLE: DES is obsolete
    DES_cblock des_key;
    DES_key_schedule schedule;
    memcpy(des_key, key.c_str(), 8);
    DES_set_key(&des_key, &schedule);
}

int generate_token() {
    // VULNERABLE: Non-cryptographic random
    srand(time(NULL));
    return rand();
}

class WeakPRNG {
public:
    int next() {
        // VULNERABLE: Predictable random
        return std::rand() % 1000000;
    }
};

std::string weak_session_id() {
    // VULNERABLE: Predictable session ID
    srand(time(nullptr));
    return std::to_string(rand());
}
